unlock-app/src/components/interface/locks/CheckoutUrl/elements/CheckoutPreview.tsx

```diff
@@ -0,0 +1,104 @@
+import { Button } from '@unlock-protocol/ui'
+import { useEffect, useState } from 'react'
+import { Checkout } from '~/components/interface/checkout/alpha/Checkout'
+import { selectProvider } from '~/hooks/useAuthenticate'
+import { useCheckoutCommunication } from '~/hooks/useCheckoutCommunication'
+import { PaywallConfig } from '~/unlockTypes'
+import { useConfig } from '~/utils/withConfig'
+import useClipboard from 'react-use-clipboard'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import FileSaver from 'file-saver'
+
+interface CheckoutPreviewProps {
+  paywallConfig?: PaywallConfig
+}
+
+const onDownloadJson = (paywallConfig: PaywallConfig) => {
+  const fileName = 'checkout-config.json'
+
+  // Create a blob of the data
+  const fileToSave = new Blob([JSON.stringify(paywallConfig)], {
+    type: 'application/json',
+  })
+
+  FileSaver.saveAs(fileToSave, fileName)
+}
+export const CheckoutPreview = ({ paywallConfig }: CheckoutPreviewProps) => {
+  const [checkoutUrl, setCheckoutUrl] = useState('')
+  const config = useConfig()
+  // Fetch config from parent in iframe context
+  const communication = useCheckoutCommunication()
+
+  const injectedProvider =
+    communication.providerAdapter || selectProvider(config)
+
+  const checkoutRedirectURI = ''
+
+  const hasLocks = Object.entries(paywallConfig?.locks ?? {})?.length > 0
+
+  useEffect(() => {
+    const url = new URL(`${window.location.origin}/alpha/checkout`)
+    url.searchParams.append(
+      'paywallConfig',
+      encodeURIComponent(JSON.stringify(paywallConfig))
+    )
+
+    if (paywallConfig?.redirectUri?.length) {
+      url.searchParams.append('redirectUri', paywallConfig.redirectUri)
+    }
+    setCheckoutUrl(url.toString())
+  }, [paywallConfig])
+
+  const [_isCopied, setCopied] = useClipboard(checkoutUrl, {
+    successDuration: 2000,
+  })
+
+  const buttonVariant = hasLocks ? 'primary' : 'outlined-primary'
+
+  return (
+    <div className="flex items-center justify-center w-full py-10 bg-gray-300 rounded-3xl">
+      <div className="flex items-center justify-center w-full max-w-lg">
+        {paywallConfig && (
+          <div className="flex flex-col items-center w-full gap-4">
+            <Checkout
+              injectedProvider={injectedProvider as any}
+              communication={communication}
+              paywallConfig={paywallConfig as any}
+              redirectURI={
+                checkoutRedirectURI ? new URL(checkoutRedirectURI) : undefined
```

Comment:
HA! I realize we never spoke about the redirect URL configuration

Reply:
we get this value actually, maybe we can move to the top position instead:

<img width="730" alt="Screenshot 2022-10-24 at 12 13 30" src="https://user-images.githubusercontent.com/20865711/197503595-620102d1-91dd-4840-b6bf-db6796255a3e.png">

Reply:
> we get this value actually, maybe we can move to the top position instead:

Yes let's do this please.

Reply:
moved to first position for the section

<img width="702" alt="Screenshot 2022-10-24 at 16 03 25" src="https://user-images.githubusercontent.com/20865711/197545271-b2e21082-e1a2-46c3-82c5-53b15e249579.png">